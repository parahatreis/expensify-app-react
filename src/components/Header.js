import React from 'react'
import { NavLink } from 'react-router-dom'


const Header = () => (
   <header>
      <h1>Expensify</h1>
      <NavLink activeClassName="selected" exact to="/">Dashboard</NavLink>
      <NavLink activeClassName="selected" to="/create">Create Expense</NavLink>
   </header>
);

export default Header