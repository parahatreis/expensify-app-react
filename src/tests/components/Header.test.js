import React from 'react';
// import ReactShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';
// import toJSON from 'enzyme-to-json';
import Header from '../../components/Header';


test('render Header test',() => {
   const wrapper = shallow(<Header />);
   expect(wrapper).toMatchSnapshot();
   // const renderer = new ReactShallowRenderer();
   // renderer.render(<Header />);
   // expect(renderer.getRenderOutput()).toMatchSnapshot();
});