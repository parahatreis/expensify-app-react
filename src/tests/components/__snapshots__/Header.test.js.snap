// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Header test 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <header>
    <h1>
        Expensify
    </h1>
    <NavLink
        activeClassName="selected"
        exact={true}
        to="/"
    >
        Dashboard
    </NavLink>
    <NavLink
        activeClassName="selected"
        to="/create"
    >
        Create Expense
    </NavLink>
</header>,
  "nodes": Array [
    <header>
      <h1>
            Expensify
      </h1>
      <NavLink
            activeClassName="selected"
            exact={true}
            to="/"
      >
            Dashboard
      </NavLink>
      <NavLink
            activeClassName="selected"
            to="/create"
      >
            Create Expense
      </NavLink>
</header>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_didScheduleRenderPhaseUpdate": false,
    "_dispatcher": Object {
      "readContext": [Function],
      "useCallback": [Function],
      "useContext": [Function],
      "useDebugValue": [Function],
      "useDeferredValue": [Function],
      "useEffect": [Function],
      "useImperativeHandle": [Function],
      "useLayoutEffect": [Function],
      "useMemo": [Function],
      "useReducer": [Function],
      "useRef": [Function],
      "useResponder": [Function],
      "useState": [Function],
      "useTransition": [Function],
    },
    "_element": <Header />,
    "_firstWorkInProgressHook": null,
    "_forcedUpdate": false,
    "_instance": null,
    "_isReRender": false,
    "_newState": null,
    "_numberOfReRenders": 0,
    "_renderPhaseUpdates": null,
    "_rendered": <header>
      <h1>
            Expensify
      </h1>
      <NavLink
            activeClassName="selected"
            exact={true}
            to="/"
      >
            Dashboard
      </NavLink>
      <NavLink
            activeClassName="selected"
            to="/create"
      >
            Create Expense
      </NavLink>
</header>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "_workInProgressHook": null,
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Header />,
}
`;
